pkgs/top-level/all-packages.nix

```diff
@@ -35972,6 +35972,15 @@ with pkgs;
 
   vkeybd = callPackage ../applications/audio/vkeybd { };
 
+  vlc-bin = vlc-bin-macos.override {
+    variant = if (stdenv.isDarwin && stdenv.isAarch64)
+              then "arm64"
+              else if (stdenv.isDarwin && stdenv.isx86_64)
+              then "intel64"
+              else (throw "Unsupported system: ${stdenv.hostPlatform.system}");
```

Comment:
I'd move that logic to `pkgs/by-name/vl/vlc-bin-macos/package.nix` (but is the `-macos` suffix really necessary? The pattern seems unprecedented in Nixpkgs.)

Reply:
I wanted to make it clear that it is a Macos-only binary.